### ISSUE TITLE
TFS GitVersion Task 5.0.1 does not use GitVersion.yml by default

### ISSUE THREAD

**Issue by roryprimrose** [GitTools/GitVersion#1867]
I have installed the GitVersion build task on a TFS 2018 R2 instance then ran it against a repo that contains a GitVersion.yml in the repo root.

```
mode: ContinuousDeployment
branches:
  master:
    tag: beta
  hotfix:
    tag: useBranchName
```

The task config (exported) looks like this:

```
   {
      "displayName": "GitVersion",
      "alwaysRun": false,
      "continueOnError": false,
      "condition": "succeeded()",
      "enabled": true,
      "timeoutInMinutes": 0,
      "inputs": {
        "runtime": "core",
        "preferBundledVersion": "true",
        "configFilePath": "",
        "updateAssemblyInfo": "false",
        "updateAssemblyInfoFilename": "",
        "gitVersionPath": "",
        "targetPath": "",
        "additionalArguments": ""
      },
      "task": {
        "id": "e5383830-3f75-11e5-82ed-81492570a08e",
        "versionSpec": "5.*",
        "definitionType": "task"
      }
    },
```

Running the task without the config path set will generate the wrong pre-release tags. The master branch would not contain a prerelease tag, hotfix would use `beta` which is the default OOTB config. 

Seems like the default loading of GitVersion.yml isn't happening. The code for the task looks fine so I'm assuming it is an issue with the GitVersionCore logic.

The workaround is to specifically set the config path property on the task to GitVersion.yml. The build now picks up the config and assigns the correct prerelease tags. For example:

```
   {
      "displayName": "GitVersion",
      "alwaysRun": false,
      "continueOnError": false,
      "condition": "succeeded()",
      "enabled": true,
      "timeoutInMinutes": 0,
      "inputs": {
        "runtime": "core",
        "preferBundledVersion": "true",
        "configFilePath": "GitVersion.yml",
        "updateAssemblyInfo": "false",
        "updateAssemblyInfoFilename": "",
        "gitVersionPath": "",
        "targetPath": "",
        "additionalArguments": ""
      },
      "task": {
        "id": "e5383830-3f75-11e5-82ed-81492570a08e",
        "versionSpec": "5.*",
        "definitionType": "task"
      }
    },
```

**Comment by arturcic** [comment]
@roryprimrose have you tried this extension https://marketplace.visualstudio.com/items?itemName=gittools.usegitversion-preview?

**Comment by roryprimrose** [comment]
Haven’t yet. I don’t control that instance so will need to request an admin to install it.

**Comment by arturcic** [comment]
let me know if it worked with this extension

**Comment by thoemmi** [comment]
I'm using the GitVersion task in Azure DevOps as well, and experience the same issue.
To me it looks like if you **don't** specify the config file, it passes the build agent's work folder for `/config` instead of omitting the option. Here's the relevant line from the build log:

`"C:\Program Files\dotnet\dotnet.exe" E:\vsts\agent-1\_work\_tasks\GitVersion_e5983830-3f75-11e5-82ed-81492570a08e\5.0.1\core\GitVersion.dll E:\vsts\agent-1\_work\15\s /output buildserver /nofetch /config E:\vsts\agent-1\_work`

**Comment by thoemmi** [comment]
I checked the Preview extension, which seems to fix the issue:
`E:\vsts\agent-1\_work\_tool\GitVersion.Tool\5.0.1\x64\dotnet-gitversion.exe E:/vsts/agent-1/_work/15/s /output buildserver /nofetch`

No `/config` option is passed if none is specified ✔

However, I think it's awkward that I have to specify a version spec (I've used `5.x`)

**Comment by arturcic** [comment]
we'll I do understand that, but using the new extension allow to use older version, or preview versions without waiting the stable release. Previously you could use only the version bundled in the extension

**Comment by thoemmi** [comment]
I get your point, however it's strange from a user's point.
Also it's disturbing that the preview extension uses the same name as the official release. When editing a pipeline, there now two tasks named _GitVersion Task_. Have you considered to add a " Preview" suffix? Additionally you could tag the extension as Preview, see [Mark an extension to be in preview](https://docs.microsoft.com/en-us/azure/devops/extend/develop/manifest?view=azure-devops#mark-an-extension-to-be-in-preview).

**Comment by arturcic** [comment]
A PR is very welcomed, if you could send a PR we could improve it. As for the versionSpec. It's similar to how Microsoft did for `UseDotNet` or `UseRuby` or other `Use*` tasks. So it's common structure, that's what I had as inspiration for the change.

**Comment by arturcic** [comment]
You can check it here https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/tool/dotnet-core-tool-installer?view=azure-devops

**Comment by thoemmi** [comment]
I have created a PR to add ` (Preview)` to the task's name.

Regarding the _preview_ tag for the extension: I was wrong, the tag is already there ;)

**Comment by arturcic** [comment]
Closed by #1871
